src/Catrobat/Controller/Web/NotificationsController.php

```diff
@@ -0,0 +1,216 @@
+        $notification_instance[$notification->getId()] = 'remix';
+      }
+      if (null == $notification_instance[$notification->getId()])
+      {
+        $notification_instance[$notification->getId()] = 'other';
+      }
+      if (null !== $user)
+      {
+        $avatar = $user->getAvatar();
+        if ($avatar)
+        {
+          $avatars[$notification->getId()] = $avatar;
+        }
+      }
+      if ($notification->getSeen() && $found_notification)
+      {
+        $old_notifications[$notification->getId()] = $notification;
+      }
+      elseif (!$notification->getSeen() && $found_notification)
+      {
+        $new_notifications[$notification->getId()] = $notification;
+      }
+    }
+    $response = $this->render('Notifications/notifications.html.twig', [
+      'oldNotifications' => $old_notifications,
+      'newNotifications' => $new_notifications,
+      'avatars' => $avatars,
+      'allNotifications' => $all_notifications,
+      'followerNotifications' => $follower_notifications,
+      'commentNotifications' => $comment_notifications,
+      'reactionNotifications' => $reaction_notifications,
+      'remixNotifications' => $remix_notifications,
+      'instance' => $notification_instance,
+    ]);
+    $response->headers->set('Cache-Control', 'no-store, must-revalidate, max-age=0');
+    $response->headers->set('Pragma', 'no-cache');
+
+    return $response;
+  }
+
+  /**
+   * @Route("/user_notifications/markasread/{notification_id}", name="notification_mark_as_read",
+   * requirements={"notification_id": "\d+"}, defaults={"notification_id": null}, methods={"GET"})
+   */
+  public function markNotificationAsRead(int $notification_id,
+                                              CatroNotificationService $notification_service,
+                                              CatroNotificationRepository $notification_repo): JsonResponse
+  {
+    $user = $this->getUser();
+    if (null === $user)
+    {
+      return JsonResponse::create(['success' => false, 'message' => 'User not logged in']);
+    }
+    $notification_seen = $notification_repo->findOneBy(['id' => $notification_id, 'user' => $user]);
+    if (null === $notification_seen)
+    {
+      return new JsonResponse(['success' => false,
+        'message' => 'Notification not found or doesnt belong to user', ]);
+    }
+    $notification_service->markSeen([$notification_seen]);
+
+    return new JsonResponse(['success' => true]);
+  }
+
+  /**
+   * @Route("/user_notifications/notifications/count", name="notifications_count", methods={"GET"})
+   */
+  public function notificationsCountAction(CatroNotificationRepository $notification_repo,
+                                               RemixManager $remix_manager): JsonResponse
+  {
+    /** @var User|null $user */
+    $user = $this->getUser();
+    if (!$user)
+    {
+      return JsonResponse::create(['statusCode' => StatusCode::LOGIN_ERROR]);
```

Comment:
A HTTP respsone has already a status code, no need to create a second one in the body